Model/SourcesRepository.php

```diff
@@ -0,0 +1,147 @@
+use Ampersand\DisableStockReservation\Model\SourcesFactory;
+use Ampersand\DisableStockReservation\Model\ResourceModel\Sources;
+use Ampersand\DisableStockReservation\Model\Sources as SourceModel;
+use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionResultInterfaceFactory;
+use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionResultInterface;
+use Ampersand\DisableStockReservation\Service\SourcesConverter;
+use Ampersand\DisableStockReservation\Model\ResourceModel\Sources\CollectionFactory;
+
+/**
+ * Class SourcesRepository
+ * @package Ampersand\DisableStockReservation\Model
+ */
+class SourcesRepository
+{
+    /**
+     * @var SourcesFactory
+     */
+    protected $sourcesFactory;
+
+    /**
+     * @var Sources
+     */
+    protected $sourcesResourceModel;
+
+    /**
+     * @var SourceSelectionResultInterfaceFactory
+     */
+    private $sourceSelectionResultFactory;
+
+    /**
+     * @var SourcesConverter
+     */
+    private $sourcesConverter;
+
+    /**
+     * @var CollectionFactory
+     */
+    private $collectionFactory;
+
+    /**
+     * SourcesRepository constructor.
+     * @param SourcesFactory $sourcesFactory
+     * @param Sources $sourcesResourceModel
+     * @param SourceSelectionResultInterfaceFactory $sourceSelectionResultFactory
+     * @param SourcesConverter $sourcesConverter
+     * @param CollectionFactory $collectionFactory
+     */
+    public function __construct(
+        SourcesFactory $sourcesFactory,
+        Sources $sourcesResourceModel,
+        SourceSelectionResultInterfaceFactory $sourceSelectionResultFactory,
+        SourcesConverter $sourcesConverter,
+        CollectionFactory $collectionFactory
+    ) {
+        $this->sourcesFactory = $sourcesFactory;
+        $this->sourcesResourceModel = $sourcesResourceModel;
+        $this->sourceSelectionResultFactory = $sourceSelectionResultFactory;
+        $this->sourcesConverter = $sourcesConverter;
+        $this->collectionFactory = $collectionFactory;
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return SourceModel
+     * @throws \Exception
+     */
+    public function getByOrderId(string $orderId): SourceModel
+    {
+        /** @var SourceModel $sourcesModel */
+        $sourcesModel = $this->sourcesFactory->create();
+        $this->sourcesResourceModel->load(
+            $sourcesModel,
+            $orderId,
+            'order_id'
+        );
+
+        if (!$sourcesModel->getId() || $sourcesModel->getId() === null) {
+            throw new NoSuchEntityException();
+        }
+
+        return $sourcesModel;
+    }
+
+    /**
+     * @return array
+     */
+    public function getOrdersSourcesCollection()
+    {
+        $ordersSourcesItems = $this->collectionFactory->create()->getItems();
+
+        $orderSources = [];
+        foreach ($ordersSourcesItems as $item) {
+            $orderSources[$item->getData('order_id')] = $item->getData('sources');
```

Comment:
As you've defined an interface for your model we should probably use the `getOrderId` methods etc as our "contract" implies that these methods are concrete and will return the values we need. No need to `getData` in this instance